src/handler/replace-document.ts

```diff
@@ -29,11 +30,15 @@ export default async (
     return {};
   }
 
-  if (data.id !== body.id) {
+  /**
+   * Falling back to `id` partitionKey for now.
+   */
+  const partitionKey = getPartitionKey(collection) || "id";
+  if (data[partitionKey] !== body[partitionKey]) {
```

Comment:
The partition key can be nested objects like `/foo/bar` for docs `{ foo: { bar } }`. It would be great if we can support these cases. (not necessary for now I think tho)